src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py

```diff
@@ -287,6 +289,43 @@ def test_skip(rule_runner: PythonRuleRunner) -> None:
     assert not result
 
 
+def test_passing_cache_clear(rule_runner: PythonRuleRunner) -> None:
```

Comment:
Note - I have a branch up called [`pyright-test-only`](https://github.com/krishnan-chandra/pants/tree/pyright-test-only) that contains only these test changes without the changes in `rules.py`.

Below are test outputs from my terminal for both branches:

**Passing:**

```bash
~/Projects/pants git:(pyright-venv-fix)

./pants test ./src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py -- -k test_passing_cache_clear
08:25:58.69 [INFO] Completed: Run Pytest - src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py:tests - succeeded.

✓ src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py:tests succeeded in 19.79s.
```

**Failing:**

```bash
~/Projects/pants git:(pyright-test-only)

./pants test ./src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py -- -k test_passing_cache_clear
There is no pantsd metadata at /Users/krishnanchandra/Projects/pants/.pants.d/pids/460a4b48a969/pantsd.
08:27:05.20 [INFO] Initializing scheduler...
08:27:18.04 [INFO] Scheduler initialized.
08:27:37.63 [ERROR] Completed: Run Pytest - src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py:tests - failed (exit code 1).
============================= test session starts ==============================
collecting ... collected 14 items / 13 deselected / 1 selected

src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py::test_passing_cache_clear FAILED [100%]

=================================== FAILURES ===================================
___________________________ test_passing_cache_clear ___________________________

rule_runner = RuleRunner(build_root=/private/tmp/_BUILD_ROOTx4cb7qsb)

    def test_passing_cache_clear(rule_runner: PythonRuleRunner) -> None:
        # Ensure that the requirements venv must be created, by adding in a third-party
        # requirement to the test code.
        rule_runner.write_files(
            {
                "BUILD": "python_requirement(name='more-itertools', requirements=['more-itertools==8.4.0'])",
                f"{PACKAGE}/f.py": dedent(
                    """\
                from more_itertools import is_sorted

                assert is_sorted([1, 2, 3]) is True
                """
                ),
                f"{PACKAGE}/BUILD": "python_sources()",
            }
        )
        tgt = rule_runner.get_target(Address(PACKAGE, relative_file_path="f.py"))

        # On the first run, it should work as advertised with no modifications.
        with temporary_dir() as cache_dir:
            result = run_pyright(rule_runner, [tgt], extra_args=[f"--named-caches-dir={cache_dir}"])
            assert len(result) == 1
>           assert result[0].exit_code == 0
E           assert 1 == 0
E             +1
E             -0

src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py:314: AssertionError
- generated xml file: /private/var/folders/0w/zhpm9fs12zzcmt3m5799xb9c0000gn/T/pants-sandbox-h7VtEl/src.python.pants.backend.python.typecheck.pyright.rules_integration_test.py.tests.xml -
=========================== short test summary info ============================
FAILED src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py::test_passing_cache_clear
====================== 1 failed, 13 deselected in 14.97s =======================



✕ src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py:tests failed in 15.48s.
```